src/v/kafka/server/replicated_partition.cc

```diff
@@ -617,23 +617,41 @@ result<partition_info> replicated_partition::get_partition_info() const {
     if (followers.has_error()) {
         return followers.error();
     }
-    auto start_offset = _partition->raft_start_offset();
 
-    auto clamped_translate = [this, start_offset](model::offset to_translate) {
-        return to_translate >= start_offset
-                 ? _translator->from_log_offset(to_translate)
-                 : _translator->from_log_offset(start_offset);
-    };
-
-    for (const auto& follower_metric : followers.value()) {
-        ret.replicas.push_back(replica_info{
-          .id = follower_metric.id,
-          .high_watermark = model::next_offset(
-            clamped_translate(follower_metric.match_index)),
-          .log_end_offset = model::next_offset(
-            clamped_translate(follower_metric.dirty_log_index)),
-          .is_alive = follower_metric.is_live,
-        });
+    /**
+     * Special handling of read replicas.
+     *
+     * Read replica replicas have all the same high watermark as the leader
+     */
```

Comment:
I'm not sure if it's important for API correctness, but this statement is generally not true. Read replicas followers may lag, in which case the view of the remote segments they see can be stale. There isn't an easy way to get all follower RRRs' actual HWMs AFAIK.

Are the HWM and end offsets here used by the client, or is this only used for server-side replica selection and indicates that each read replica should be selected with equal weight?

If the latter, it would be great to document the correctness flexibility/requirements of `get_partition_info()` somewhere.

Reply:
the offsets here are only used to determine which replica to use to serve the reads. I will add a comment on the method

Reply:
It is still bad if we route the client to the replica that doesn't have the offsets the client needs, right?